Added target information

PoiDoe opened the pull request:
shows details about currently tarrgeted mob

info window only shows when local char has target or if the main kanan window is open

the codes super messy as a i made it in the middle of the night while high but it works

https://user-images.githubusercontent.com/20039012/147513618-958e53db-ac6c-40c5-bc44-de079a7211b7.mp4

cursey commented on Kanan/Currtarget.cpp:
```diff
@@ -0,0 +1,256 @@
+
+
+			//check over all the enitys and find the one being targeted
+			for (auto& character : m_characters) {
```
This loop seems unnecessary, you already have the target stored in `target`

cursey commented on Kanan/Currtarget.hpp:
```diff
@@ -0,0 +1,34 @@
+		bool showpos{};
+		bool showothertarget{};
+		bool showtargethp{};
+		std::forward_list<KCharacter*> m_characters;
```
`m_characters` seems unnecessary because you can (and already do) get the target via `Game::getCharacterByID`.

cursey commented on Kanan/Currtarget.cpp:
```diff
@@ -0,0 +1,256 @@
+
+			//create a list of all local chars
+
+			for (uint32_t i = 0; i <= highestIndex && node != nullptr; ++i, node = node->next) {
```
This loop seems unnecessary, you've already gotten the target (`target`) and don't use `m_characters` for anything else.

cursey commented on Kanan/Currtarget.cpp:
```diff
@@ -0,0 +1,256 @@
+			target = g_kanan->getGame()->getCharacterByID(localCharacter->targetID);
+			if (target == nullptr) { hastarget = false; }
+			else { hastarget = true; }
+		}
```
Could be replaced with clearer logic. Maybe something like:
```
hastarget = localCharacter && localCharacter->targetID != 0;
```

PoiDoe replied:
not sure how to do this cleaner, i only want to checking for target if local player exists

cursey commented on Kanan/Currtarget.hpp:
```diff
@@ -0,0 +1,34 @@
+		bool hastarget{};
+		bool showpos{};
+		bool showothertarget{};
+		bool showtargethp{};
```
These members could be prefixed with `m_` to match the rest of Kanan's styling.

cursey reviewed (commented):
I like the idea, would appreciate it if the code was just touched up a bit.

PoiDoe commented:
Updated members to match kanans styling

removed unneeded loops.
unsure how to clean up logic currently in ::onframe() that is beond my skill level atm

cursey commented:
Thanks for the updates!
